source/adapters/hip/adapter.cpp

```diff
@@ -10,14 +10,44 @@
 
 #include "adapter.hpp"
 #include "common.hpp"
+#include "logger/ur_logger.hpp"
 
 #include <atomic>
 #include <ur_api.h>
 
 struct ur_adapter_handle_t_ {
   std::atomic<uint32_t> RefCount = 0;
+  logger::Logger &logger;
+  ur_adapter_handle_t_();
 };
 
+class ur_legacy_sink : public logger::Sink {
+public:
+  ur_legacy_sink(std::string logger_name = "", bool skip_prefix = true)
+      : Sink(std::move(logger_name), skip_prefix) {
+    this->ostream = &std::cerr;
```

Comment:
Taking the address of `std::cerr` seems like a strange thing to do. Why are we doing this?

Reply:
This is part of abstraction - in logger we use abstract class 'sink', which can write to any kind of output stream. in this legacy sink we are writing to stderr, so we have to use cerr stream.